tests/Tokenizer/TokensTest.php

```diff
@@ -1174,6 +1174,26 @@ public function provideRemoveTrailingWhitespaceCases()
         return $cases;
     }
 
+    public function testRemovingLeadingWhitespaceWithEmptyTokenInCollection()
+    {
+        $tokens = Tokens::fromCode("<?php\n    /* I will be removed */MY_INDEX_IS_THREE;foo();");
+        $tokens->clearAt(2);
+
+        $tokens->removeLeadingWhitespace(3);
+
+        $this->assertSame("<?php\nMY_INDEX_IS_THREE;foo();", $tokens->generateCode());
+    }
+
+    public function testRemovingTrailingWhitespaceWithEmptyTokenInCollection()
+    {
+        $tokens = Tokens::fromCode("<?php\nMY_INDEX_IS_ONE/* I will be removed */    ;foo();");
+        $tokens->clearAt(2);
+
+        $tokens->removeTrailingWhitespace(1);
+
+        $this->assertSame("<?php\nMY_INDEX_IS_ONE;foo();", $tokens->generateCode());
```

Comment:
can we use `assertTokens` instead ? I would like to ensure that output collection is proper

Reply:
like this, but use `clearEmptyTokens` on `$tokens`